github-actions/scan/src/input-helper.ts

```diff
@@ -16,3 +19,62 @@ export function split(input: string): string[] {
         .map(item => item.trim())
         .filter(item => item.length > 0);
 }
+
+/**
+ * This method checks the default environment variables 'http_proxy' and 'https_proxy' for http/https proxy specification.
+ * The variables are expected to be specified in this format: http_proxy='http://user:password@proxy.example.org:1234'.
+ * This results in an URL object like this:
+ * URL {
+      href: 'http://user:password@proxy.example.org:1234/',
+      origin: 'http://proxy.example.org:1234',
+      protocol: 'http:',
+      username: 'user',
+      password: 'password',
+      host: 'proxy.example.org:1234',
+      hostname: 'proxy.example.org',
+      port: '1234',
+      pathname: '/',
+      search: '',
+      searchParams: URLSearchParams {},
+      hash: ''
+    }
+ * @returns configured AxiosProxyConfig or undefined if no proxy was
+ * @throws error if the proxy URL inside the ENV variable is an invalid URL
+ */
+export function resolveProxyConfig(): AxiosProxyConfig | undefined {
+    const httpsProxy = process.env.https_proxy || undefined;
+    const httpProxy = process.env.http_proxy || undefined;
```

Comment:
I am asking myself if this wouldn't be just enough:

```suggestion
    const httpsProxy = process.env.https_proxy;
    const httpProxy = process.env.http_proxy;
```